File: 扩大点击范围/扩大点击范围/ViewController.m
//
//  ViewController.m
//  扩大点击范围
//
//  Created by laiyp on 2018/7/24.
//  Copyright © 2018年 laiyongpeng. All rights reserved.
//

#import "ViewController.h"
#import "UIView+ExpandScope.h"
#import "UIButton+YPBlock.h"
#import "FirstVC.h"
#import "UIView+Extension.h"

@interface ViewController ()
@property (nonatomic, strong) FirstVC *vc;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    UIView *view = [[UIView alloc]initWithFrame:CGRectMake(100, 50, 100, 100)];
    [view lyp_addRounderCornerWithRadius:10.0 size:CGSizeMake(view.width, view.height)];
    view.backgroundColor = [UIColor grayColor];
    view.touchSize = CGSizeMake(200, 200);
    view.clipsToBounds = YES;
    [self.view addSubview:view];
    
    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc]initWithTarget:self action:@selector(tap)];
    [view addGestureRecognizer:tap];
    
    
    UIButton *button = [[UIButton alloc]init];
    button.frame = CGRectMake(100, 350, 200, 100);
    button.backgroundColor = [UIColor greenColor];
    __weak typeof(self) weakself = self;
    __weak __typeof__(button) weakButton = button;
    [button yp_addEventHandler:^{
        weakButton.selected = !weakButton.selected;
        if (weakButton.selected) {
            [weakButton setTitle:@"黄色" forState:UIControlStateNormal];
            weakself.view.backgroundColor = [UIColor yellowColor];
        }else{
            [weakButton setTitle:@"绿色" forState:UIControlStateNormal];
            weakself.view.backgroundColor = [UIColor whiteColor];
        }
    } forControlEvents:UIControlEventTouchUpInside];
//    [self.view addSubview:button];
    
    
    FirstVC *vc = [[FirstVC alloc]init];
    vc.view.frame = CGRectMake(100, 200, 300, 300);
    vc.view.backgroundColor = [UIColor yellowColor];
    self.vc = vc;
    [self.view addSubview:vc.view];
}

-(void)tap{
    
    NSLog(@"%@",self.vc.addButton.allTargets);
    
//   self performSelector: withObject:<#(nullable id)#> afterDelay:<#(NSTimeInterval)#>
    
}

@end

File: Swift/swift5.0/swift5.0/ViewController.swift
//
//  ViewController.swift
//  swift5.0
//
//  Created by 赖永鹏 on 2019/8/9.
//  Copyright © 2019年 LYP. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    class Size{
        var width: Int
        var height: Int
        init(width: Int, height: Int) {
            self.width = width
            self.height = height
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
//   // 值类型赋值是将内容重新拷贝一份，属于深拷贝
//        struct Point {
//            var x: Int
//            var y: Int
//        }
//
//        let p1 = Point(x: 10, y: 20)
//        let p2 = p1
        
        test()
        
    }
  

    func test(){
        // 指针类型赋值 是将指针重新拷贝一份，指向的是d同一份内存，所以是浅拷贝
        let s1 = Size(width: 10, height: 20)
        let s2 = s1
        s2.width = 30
        
        print("s1.width = \(s1.width) s2.width = \(s2.width)  ")
    }
}

